tokio-process/src/lib.rs

```diff
@@ -501,24 +523,49 @@ impl Child {
     /// `stderr(Stdio::piped())`, respectively, when creating a `Command`.
     pub fn wait_with_output(mut self) -> WaitWithOutput {
         drop(self.stdin().take());
-        let stdout = match self.stdout().take() {
-            Some(io) => Either::A(read_to_end(io, Vec::new()).map(|p| p.1)),
-            None => Either::B(ok(Vec::new())),
+        let stdout_val = self.stdout.take();
+        let stderr_val = self.stderr.take();
+        let stdout_fut = async {
+            match stdout_val {
+                Some(mut io) => {
+                    let mut vec = Vec::new();
+                    dbg!("READ TO END START");
```

Comment:
Might be good to omit the debug code before we merge. We could eventually use `tracing` here for debugging but im not sure if it's truly necessary yet